scripts/env-check.sh

```diff
@@ -0,0 +1,79 @@
+#!/usr/bin/env bash
+
+set -e
+
+if [[ "$#" -lt 1 ]]; then
+    echo "Usage: $(basename "$0") <repo> [env_prefix (default: NSM_)]"
+    exit 1
+fi
+
+repo="$(readlink -e "$1")"
+env_prefix="${2:-NSM_}"
+cmd_name="cmd"
+
+
+is_in_order() {
+    local -a arr
+    mapfile -t arr < <(echo -en "${1}\n${2}" | sort)
+
+    if [[ "${arr[0]}" == "$1" ]]; then
+        return 0
+    fi
+    return 1
+}
+
+print_mismatch() {
+    echo "Mismatch, no ${1} environment variable found in ${2}"
+}
+
+build_cmd() {
+    local -r repo="$1"
+    local -r output="$2"
+    pushd . &>/dev/null
+    cd "$repo"
+    go build -o "$output" .
+    popd &>/dev/null
+}
```

Comment:
Yeah, my though here was that it would be more convenient to have the script do the build, and that way you have a contained script that can be called on any repo manually without the job, but I don't really have any strong feelings about it.
I guess in the unit_test approach its a bit cleaner to have this in the job itself.